### PR TITLE
Add EffectButtonStyle

### PR DESCRIPTION
**Pull request by kimgaeunnn**
### Added
- Add effect which reacts to button pressing action

### Changed
- Apply `EffectButtonStyle` to copy button
- Commonize CopyButton on CoralUI

### Checklist ✅

- [x] The code has been linted using the command `make lint`.
- [x] Unit tests have been written to ensure major functionality is not broken.

### CODE REVIEW

File: Projects/Common/CoralUI/Sources/CopyButton.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                .imageScale(.large)
+                .foregroundColor(Color.white)
+
+            Text("Copied!")
+                .font(Font.body.bold())
+                .transition(.move(edge: .trailing).combined(with: .opacity))
+                .foregroundColor(.white)
+        }
+    }
+}
```
Comment (DevYeom) [inline Projects/Common/CoralUI/Sources/CopyButton.swift:38]: Please add a newline between 37 and 38

File: Projects/Common/CoralUI/Sources/TextEffectButtonStyle.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// The MIT License (MIT)
+//
+// https://github.com/DevYeom/Coral
+
+import SwiftUI
+
+public struct EffectButtonStyle<Content: View>: ButtonStyle {
+    let effectView: Content
```
Comment (DevYeom) [inline Projects/Common/CoralUI/Sources/TextEffectButtonStyle.swift:8]: Please add a newline between 7 and 8

**Review by DevYeom — APPROVED** [review]

File: Projects/Common/CoralUI/Sources/CopyButton.swift (hunk 1 above)
Comment (kimgaeunnn) [reply Projects/Common/CoralUI/Sources/CopyButton.swift]: thanks! checkout new commit plz~ 5456261c9b8d353c0261b459a29f782fcf50bf49

File: Projects/Common/CoralUI/Sources/TextEffectButtonStyle.swift (hunk 2 above)
Comment (kimgaeunnn) [reply Projects/Common/CoralUI/Sources/TextEffectButtonStyle.swift]: thanks! checkout new commit plz~ 5456261c9b8d353c0261b459a29f782fcf50bf49

**Pull request merged**